Feature/ab#15 events

Joren-Thijs opened the pull request:
Fixed all issues raised during the demo.
And added an events location service to use.

Joren-Thijs committed:
feat(events) fixed table date formats
and registrants display

AB#49

Joren-Thijs committed:
feat(events) event overvie location optimized

AB#17

Joren-Thijs committed:
feat(events) added row higlighting
if max number of registrants has been exceeded

AB#58

Joren-Thijs committed:
feat(events) created singerEventLocation service

and proxy

AB#16

vyruz1986 commented on Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-overview/singerevent-overview.component.html:
```diff
@@ -16,9 +16,10 @@
          <th mat-header-cell *matHeaderCellDef mat-sort-header>Locatie</th>
          <td mat-cell *matCellDef="let row">
-            <h5>{{ row.location.name }}</h5>
+            <p></p>
```
Why is this extra paragraph in here?

vyruz1986 commented on Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-overview/singerevent-overview.component.ts:
```diff
@@ -82,6 +82,11 @@ export class SingerEventOverviewComponent implements OnInit, AfterViewInit {
 
+   // Returns true if the max number of registrants for the event have been exceeded
+   getMaxRegistrantsExceeded(row: SingerEvent):boolean {
+      return row.currentRegistrants > row.maxRegistrants ? true : false;
```
This can be simplified to:
`return row.currentRegistrants > row.maxRegistrants;` because the return value of that statement will be `true` or `false`

BerendWouters replied:
I would call this method `isMaxRegistrantsExceeded`

vyruz1986 commented on Singer.API/ClientApp/src/app/modules/admin/components/singerevents/singerevent-overview/singerevent-overview.component.html:
```diff
@@ -127,6 +132,7 @@ <h5>{{ row.location.name }}</h5>
          mat-row
          *matRowDef="let row; columns: displayedColumns"
          class="selectable"
+         [ngClass]="{'maxRegistrantsExceeded': getMaxRegistrantsExceeded(row)}"
```
This applies the `maxRegistrantsExceeded` to the whole row, we first said to only apply this to the cell/contents of the cell. Was this done deliberately or not?

vyruz1986 commented on Singer.API/ClientApp/src/app/modules/core/services/singerevents-api/singerevent-location.proxy.ts:
```diff
@@ -0,0 +1,53 @@
+      updateSingerEventLocationDTO: UpdateSingerEventLocationDTO
+   ) {
+      return this.apiService
+         .put(`api/event/${id}`, updateSingerEventLocationDTO)
```
this should use `api/eventlocation/${id}`

vyruz1986 commented on Singer.API/ClientApp/src/app/modules/core/services/singerevents-api/singerevent-location.proxy.ts:
```diff
@@ -0,0 +1,53 @@
+      createSingerEventLocationDTO: CreateSingerEventLocationDTO
+   ): Observable<SingerEventLocationDTO> {
+      return this.apiService
+         .post('api/event', createSingerEventLocationDTO)
```
this should use `api/eventlocation/${id}`

Joren-Thijs committed:
feat(events) implemented singerEventLocationService
 in details form

AB#15

Joren-Thijs committed:
feat(events) fixed PR requests

AB#17

Joren-Thijs committed:
feat(events) renamed method

getMaxRegistrantsExceeded -> isMaxRegistrantsExceeded

AB#17

vyruz1986 commented on Singer.API/ClientApp/src/app/modules/core/services/singerevents-api/singerevent-location.proxy.ts:
```diff
@@ -0,0 +1,53 @@
+      createSingerEventLocationDTO: CreateSingerEventLocationDTO
+   ): Observable<SingerEventLocationDTO> {
+      return this.apiService
+         .post('apilocation/event', createSingerEventLocationDTO)
```
There should be a slash between 'api' and 'location' in `apilocation/event`

vyruz1986 commented:
@comediehero please also resolve conflicts (after merge of #42 )

BWouters committed:
Merge branch 'develop' into feature/AB#15-events

BWouters committed:
fix proxy url and loading of dependecies

AB#15

BWouters committed:
fix dto and event form handling (for location and agegroup)

AB#15

BWouters committed:
Merge remote-tracking branch 'origin/develop' into feature/AB#15-events

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json
#	Singer.API/ClientApp/src/app/modules/admin/admin.module.ts

BWouters committed:
build fix post merge issues

AB#15

BerendWouters commented:
Updated this PR for a bit:
* Location property in the form is now updated correctly in the backend on creation
* Updated the agegroups to be handled correctly in the form too.
* Both the previous values should be set correctly on loading existing events

Missing pieces:
* Creation of an event misses some crucial validation: end dates cannot be before startdates, if certain dropdowns are set to 'No', there should be no validation on the following fields (they should maybe even ben greyed out)
* The order of the dates is not changed yet.
* Updating an existing event fails due to broken DTO handling.

Back to you @comediehero :)

vyruz1986 committed:
Merge branch 'develop' into feature/AB#15-events

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json
#	Singer.API/Profiles/EventProfile.cs

vyruz1986 committed:
fix(Events): Minor improvements in events UI

Removed current registrants from details component
Hide before/after start/end times when corresponding day care is not
enabled

vyruz1986 committed:
chore: Added TODO for daycare start/end fields

vyruz1986 committed:
fix(Events): Finished events UI implementation

Fixed mappings to start/end time fields (by use of 3rd party timepicker)
Resized event details dialog
Made sure that add and edit DTO's map correctly to API versions

vyruz1986 committed:
Merge remote-tracking branch 'origin/develop' into feature/AB#15-events

# Conflicts:
#	Singer.API/ClientApp/package-lock.json
#	Singer.API/ClientApp/package.json
#	Singer.API/ClientApp/src/app/material.module.ts
#	Singer.API/Profiles/EventProfile.cs

BWouters committed:
feat make nullable datetime fields really nullable

AB#15

BWouters committed:
feat adds ModelState validation check for Create-route

AB#15

BerendWouters reviewed (approved):
Many formatting changes, rather hard to check the major changes.
